src/main/java/com/example/Library/services/UserDetailsServiceImpl.java

```diff
@@ -13,16 +12,17 @@
 @Service("userDetailsServiceImpl")
 public class UserDetailsServiceImpl implements UserDetailsService {
 
-    private final UserRepository userRepository;
+
+    private final UserService userService;
```

Comment:
Services could have dependency on repository, no need to replace with UserService for now